src/components/actions/extendApplicationRoute.ts

```diff
@@ -0,0 +1,67 @@
+import { combineRequestParams } from '../utilityFunctions'
+import { PermissionPolicyType, Trigger } from '../../generated/graphql'
+import { DateTime } from 'luxon'
+import DBConnect from '../databaseConnect'
+import { getPermissionNamesFromJWT } from '../data_display/helpers'
+
+export const routeExtendApplication = async (request: any, reply: any) => {
+  const { applicationId, eventCode, extensionTime, data } = combineRequestParams(request, 'camel')
+
+  // Check permissions first -- currently we just check user has ANY Reviewing
+  // permissions for current template.
+  // TO-DO: define specific permission names required for extending deadlines
+  const { permissionNames: userPermissions } = await getPermissionNamesFromJWT(request)
+
+  const templatePermissions = (
+    await DBConnect.getTemplatePermissionsFromApplication(Number(applicationId))
+  )
+    .filter(
+      (permission: any) =>
+        permission?.permissionName?.permissionPolicy?.type === PermissionPolicyType.Review
+    )
+    .map((permission: any) => permission?.permissionName?.name)
+
+  const hasPermission = userPermissions.some((permission) =>
+    templatePermissions.includes(permission)
+  )
+
+  if (!hasPermission) return reply.send({ success: false, message: 'Unauthorized' })
```

Comment:
I think it might be better to use `permissions_all` view ? We probably want to also filter by org

Reply:
This does ultimately fetch from `permissions_all` and includes "org" restrictions.

`hasPermissions` checks `userPermissions`, which comes from `getPermissionNamesFromJWT()`, which calls database function `getUserOrgPermissionNames`, which looks up the `permissions_all` table :)

```
SELECT "permissionNameId" as id,
      "permissionName" FROM permissions_all
      WHERE ${userMatch}
      AND ${orgMatch}`
```

Reply:
ok, great, thanks for clarifying, I might try and reduce/consolidate this in row permission work, I just saw that permissions_all also has template_id, and thought it would be more direct to use it